app/src/battery.c

```diff
@@ -87,7 +87,9 @@ static void zmk_battery_timer(struct k_timer *timer) {
 K_TIMER_DEFINE(battery_timer, zmk_battery_timer, NULL);
 
 static void zmk_battery_start_reporting() {
-    k_timer_start(&battery_timer, K_NO_WAIT, K_SECONDS(CONFIG_ZMK_BATTERY_REPORT_INTERVAL));
+    if (device_is_ready(battery)) {
```

Comment:
Seems reasonable, although I'm surprised that startup even succeeds when the init function below this fails with an error code using this same check.

Reply:
I believe the early return in the init function prevents zmk_battery_start_reporting() from being called on startup.